include/swift/Sema/IDETypeChecking.h

```diff
@@ -162,7 +162,8 @@ namespace swift {
 
   class SynthesizedExtensionAnalyzer {
     struct Implementation;
-    Implementation &Impl;
+    std::unique_ptr<Implementation> Impl;
```

Comment:
Not sure about this. This technique is used in others too such as `ClangImporter`. I see `GenericSignatureBuilder` uses `unique_ptr`, so we are already inconsistent 🥴. I *personally* like `Impl` being a reference because that guarantees `Impl` is non-null. But that's not a strong opinion, so I'll leave it up to you.

Reply:
I'll probably just change it back to reduce the noise. I don't have a strong opinion either, except that for the reference I had to double check it was being deleted, where as I wouldn't for a unique_ptr 🤷